src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -935,23 +936,26 @@ void PolicyHandler::OnGetListOfPermissions(const uint32_t connection_key,
   const bool is_app_registered = NULL != app.get();
   const bool is_connection_key_valid = is_app_registered && connection_key;
 
-  const std::vector<policy::FunctionalGroupPermission> permissions =
-      is_connection_key_valid ? CollectAppPermissions(connection_key)
-                              : CollectRegisteredAppsPermissions();
+  std::vector<policy::FunctionalGroupPermission> permissions;
 
-  if (permissions.empty() && is_connection_key_valid) {
-    SDL_LOG_ERROR("No permissions found for application with connection key:"
-                  << connection_key);
-    return;
-  }
+  const bool collect_result =
+      is_connection_key_valid
+          ? CollectAppPermissions(connection_key, permissions)
+          : CollectRegisteredAppsPermissions(permissions);
 
-  MessageHelper::SendGetListOfPermissionsResponse(
-      permissions,
+  if (collect_result) {
+    MessageHelper::SendGetListOfPermissionsResponse(
+        permissions,
 #ifdef EXTERNAL_PROPRIETARY_MODE
-      policy_manager->GetExternalConsentStatus(),
+        policy_manager->GetExternalConsentStatus(),
 #endif  // EXTERNAL_PROPRIETARY_MODE
-      correlation_id,
-      application_manager_);
+        correlation_id,
+        application_manager_);
+  } else {
+    SDL_LOG_ERROR(
+        "Permissions collection failed for application with connection key:"
+        << connection_key);
```

Comment:
Should we not send an error response in this case?

Reply:
@jacobkeeler done in  01c1aa6dc2c32f07117516a982589085c16906df
I made some refactor of this scope and now we send GetListOfPermission response to HMI anyway, but with error if collection result is false;